From 90e0cef356763fd58c843aee6a3066596026a0b9 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Mon, 09 Nov 2020 18:15:42 +0200
Subject: [PATCH] [SW-1441] gaudi: security: switch to use specs rrs

Switch to use the enumerated IF RTRs RRs from specs.
This also changes some of the RRs numbers, but it does not matter now,
since we have them organized in specs, so just follow specs.

Change-Id: I760b3d0bd66729d30ebc694f01707bf7ac4c3443
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/security.c b/gaudi/security.c
index c3b7fc1..c44e0b2 100644
--- a/gaudi/security.c
+++ b/gaudi/security.c
@@ -1224,39 +1224,39 @@
 
 static void preboot_secure_hbw(void)
 {
-	/* HBM RR_0 0 - 32GB */
-	preboot_secure_hbw_rr(0, 0, 0x3FFF800000000);
+	/* HBM RR 0 - 32GB */
+	preboot_secure_hbw_rr(HBW_RR_HBM, 0, 0x3FFF800000000);
 
-	/* SRAM RR_8 0x7F_F000_0000 - 0x7F_F100_0000 (16MB) */
+	/* SRAM RR 0x7F_F000_0000 - 0x7F_F100_0000 (16MB) */
 	/*
 	 * TODO: The below (both SRAM parts) are commented out temporarily
 	 * until the dynamic SRAM RR protection is in place.
 	 */
-/*	preboot_secure_hbw_rr(8, 0x7FF0000000, 0x3FFFFF000000);
+/*	preboot_secure_hbw_rr(HBW_RR_SRAM_BGN, 0x7FF0000000, 0x3FFFFF000000);
 */
-	/* SRAM RR_9 0x7F_F100_0000 - 0x7F_F140_0000 (4MB) */
-/*	preboot_secure_hbw_rr(9, 0x7FF1000000, 0x3FFFFFC00000);
+	/* SRAM RR 0x7F_F100_0000 - 0x7F_F140_0000 (4MB) */
+/*	preboot_secure_hbw_rr(HBW_RR_SRAM_END, 0x7FF1000000, 0x3FFFFFC00000);
 */
-	/* SRAM Reduct. RR_1 0x7F_F140_0000 - 0x7F_F180_0000 (4MB) */
-	preboot_secure_hbw_rr(1, 0x7FF1400000, 0x3FFFFFC00000);
+	/* SRAM Reduct. RR 0x7F_F140_0000 - 0x7F_F180_0000 (4MB) */
+	preboot_secure_hbw_rr(HBW_RR_SRAM_R_BGN, 0x7FF1400000, 0x3FFFFFC00000);
 
-	/* SRAM Reduct. RR_2 0x7F_F180_0000 - 0x7F_F200_0000 (8MB) */
-	preboot_secure_hbw_rr(2, 0x7FF1800000, 0x3FFFFF800000);
+	/* SRAM Reduct. RR 0x7F_F180_0000 - 0x7F_F200_0000 (8MB) */
+	preboot_secure_hbw_rr(HBW_RR_SRAM_R_1, 0x7FF1800000, 0x3FFFFF800000);
 
-	/* SRAM Reduct. RR_3 0x7F_F200_0000 - 0x7F_F280_0000 (8MB) */
-	preboot_secure_hbw_rr(3, 0x7FF2000000, 0x3FFFFF800000);
+	/* SRAM Reduct. RR 0x7F_F200_0000 - 0x7F_F280_0000 (8MB) */
+	preboot_secure_hbw_rr(HBW_RR_SRAM_R_END, 0x7FF2000000, 0x3FFFFF800000);
 
-	/* STM RR_4 0x7F_F400_0000 - 0x7F_F800_0000 (64MB) */
-	preboot_secure_hbw_rr(4, 0x7FF4000000, 0x3FFFFC000000);
+	/* STM RR 0x7F_F400_0000 - 0x7F_F800_0000 (64MB) */
+	preboot_secure_hbw_rr(HBW_RR_STM, 0x7FF4000000, 0x3FFFFC000000);
 
-	/* SPI Flash RR_5 0x7F_F800_0000 - 0x7F_FA00_0000 (32MB) */
-	preboot_secure_hbw_rr(5, 0x7FF8000000, 0x3FFFFE000000);
+	/* SPI Flash RR 0x7F_F800_0000 - 0x7F_FA00_0000 (32MB) */
+	preboot_secure_hbw_rr(HBW_RR_SPI_FLASH, 0x7FF8000000, 0x3FFFFE000000);
 
-	/* Scratchpad SRAM RR_6 0x7F_FBFE_0000 - 0x7F_FBFF_0000 (64KB) */
-	preboot_secure_hbw_rr(6, 0x7FFBFE0000, 0x3FFFFFFF0000);
+	/* Scratchpad SRAM RR 0x7F_FBFE_0000 - 0x7F_FBFF_0000 (64KB) */
+	preboot_secure_hbw_rr(HBW_RR_SP_SRAM, 0x7FFBFE0000, 0x3FFFFFFF0000);
 
-	/* PCIe FW SRAM RR_7 0x7F_FBFF_0000 - 0x7F_FBFF_8000 (32KB) */
-	preboot_secure_hbw_rr(7, 0x7FFBFF0000, 0x3FFFFFFF8000);
+	/* PCIe FW SRAM RR 0x7F_FBFF_0000 - 0x7F_FBFF_8000 (32KB) */
+	preboot_secure_hbw_rr(HBW_RR_PCIE_FW, 0x7FFBFF0000, 0x3FFFFFFF8000);
 
 	/*
 	 * PCIe IF HBW RRs are for direct access from PCIe IF to PSOC domain:
